runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -63,8 +69,22 @@ protected ManagementClient getOrchestrator(Layout layout) {
             throw new WorkflowException("getOrchestrator: no available orchestrators " + layout);
         }
 
-        return runtime.getLayoutView().getRuntimeLayout(layout)
+        // Select an available orchestrator
+        ManagementClient managementClient = runtime.getLayoutView().getRuntimeLayout(layout)
                 .getManagementClient(activeLayoutServers.get(0));
+
+        for (String endpoint : activeLayoutServers) {
```

Comment:
should we shuffle this list? or we might always end with the same orchestrator...

Reply:
If the orchestrator is not available (i.e. pingSync == false) then how can this happen?
This loop will ping through the active servers set and select the first one that will respond to the ping.

Reply:
I thought we wanted to have some randomness in the selection of the orchestrator, it is true that it will pick the one which pingSync == true, but chances are that if nothing goes wrong you will always be picking the first one, instead of giving the chance to any other..

Reply:
@annym We would like to avoid randomness for simplifying debugging with deterministic behaviour.

Reply:
The randomness was introduced to discover a node that is available, using the ping achieves that and a better strategy when you have more nodes that are dead (i.e. less retries) .

Reply:
ok perfect, I thought that we wanted to make the selection somehow random.